RAD-6204 6206 Update Mango Configuration Data Source To Support New Locators and Publishers

RamKannekanti opened the pull request:
RAD-6204 Update Mango Configuration Data Source To Support New Locators

RAD-6206 Update Mango Configuration Data Source To Support Publishers

RamKannekanti committed:
RAD-6204 RAD-6206 Update Mango Configuration Data Source To Support New Locators and Publishers

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+    }
+}
+
+function processSystemMetrisLocatorType(locatorValue){
```
_processSystemMetrisLocatorType_ should be _processSystemMetricsLocatorType_

RamKannekanti replied:
Corrected

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+                return 'undefined';
+           }
+        }catch(err){
+            LOG.error('Publisher Locator testing' + ' ' +  err.message); 
```
We should only have "testing" in the logs during testing, that should not be in the final PR. Also, this method is a generic method for retrieving JSON values, it's not specific to Publishers. The error log entry should not reference Publisher. It should reference an error when retrieving a JSON value. It would also be good to include the jsonObject and path data to help with troubleshooting.

RamKannekanti replied:
corrected jsonObject and path added in log entry

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+                  }
+                }      
+           } else {
+                LOG.debug("empty object receioved."); 
```
_receioved_ should be _received_

RamKannekanti replied:
corrected

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+      return this.fetchedLocatorValue;
+} 
+
+function getValueBackup(jsonObject,path){
```
What is the purpose of this method?

RamKannekanti replied:
This method removed, it is only backup of existing method

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+      return this.fetchedLocatorValue;
+} 
+   
+function getTestJson(){
```
What is the purpose of this method? If this is from testing, it should not be included in the final PR.

RamKannekanti replied:
This method removed . It provides test json data for testing.

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+var HashMap = Java.type("java.util.HashMap");
+monitoredValuesHashList = new HashMap();
+
+getAllMonitoredValues();
```
We should call this method inside _processSystemMetricsLocatorType_ so we avoid executing it if there are no locators of this type in the data source. Also, we should use the same pattern from lines 160-165 which ensures we only call this method 1 time, even if there are multiple data points using this locator type.

RamKannekanti replied:
this is fixed

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+            continue;
+        }
+        
+        if(locatorType === 'publisher') {
```
Let's add logging locatorParam1 inside the publisher method and remove it from this method.

RamKannekanti replied:
this has been fixed.

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+}
+
+function processSystemMetrisLocatorType(locatorValue){
+    return  monitoredValuesHashList.get(locatorValue);
```
We need to provide some logging if the locatorValue is not found. All the other methods include this type of logging, this one should not be different.

RamKannekanti replied:
locatorValue not found logging info added

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+        var AbstractPublisherModelObj = RestModelMapperInstance.map(PublisherVO, AbstractPublisherModel.class, Common.getUser()); 
+        var PublisherJsonString = mapperInstance.writerWithDefaultPrettyPrinter().writeValueAsString(AbstractPublisherModelObj);
+      
+        return getPublisherLocatorTypePropertyValue(PublisherJsonString,locatorValue);
```
We need to provide some logging if the locatorValue is not found. All the other methods include this type of logging, this one should not be different.

RamKannekanti replied:
this has been fixed

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+        
+        //get the bean named restObjectMapper and make sure it is of type ObjectMapper
+        var mapperInstance = Common.getBean(ObjectMapper.class,"restObjectMapper");
+        var PublisherVO= PublisherService.get(locatorParam1); 
```
We need to add a check to confirm locatorParam1 returns a Publisher.

RamKannekanti replied:
added. please check line 280-292

aasbra commented on track-mango-details.js:
```diff
@@ -25,200 +6,374 @@ var System = Java.type('java.lang.System');
+        return getPublisherLocatorTypePropertyValue(PublisherJsonString,locatorValue);
+        
+    } catch(err){
+        LOG.error('Publisher locatorParam1: ' + locatorParam1 + ' ' +  err.message);
```
We should include the locatorValue in this log as well to help with troubleshooting.

RamKannekanti replied:
yes added

aasbra reviewed (changes requested):
I've added comments that should be addressed before the PR can be merged. Please let me know if you have any questions.

RamKannekanti committed:
RAD-6204 RAD-6206 Update Mango Configuration PR # 30 review comments fixed

aasbra committed:
Update and rename track-mango-details.js to mango-configuration-tracking-data-source.js

Several bugs were resolved and added support for a new hardware locator type which reports CPU, Memory, and HD properties.
